### PR TITLE
feat(OnyxForm): implement showError property

### PR DESCRIPTION
**Pull request by JoCa96**
Relates to #1375 

- feat(OnyxForm): implement `showError` mode
- updated all relevant FormElements to have and use `showError` prop
- created dedicated `FormProps` and `FormInjectedProps` type for easier re-use and shared description
- fix(OnyxForm): update of injected prop not working when initially using default
- enabled and configured type test from vitest for ensuring vue workaround doesn't cause issues
- fix linting issue in `useDensity` computed type

### CODE REVIEW

File: .changeset/beige-cherries-explain.md
Comment (MajaZarkova) [inline .changeset/beige-cherries-explain.md:1]: Why do we have the ripple animation changeset in this PR?

File: packages/sit-onyx/src/components/OnyxForm/OnyxForm.stories.ts
Comment (MajaZarkova) [inline packages/sit-onyx/src/components/OnyxForm/OnyxForm.stories.ts:1]: I noticed that on the input example we have a boolean
image

But this is not the case for the form component. Is this intentional?

**Review by MajaZarkova — CHANGES_REQUESTED** [review]

File: .changeset/beige-cherries-explain.md
Comment (JoCa96) [reply .changeset/beige-cherries-explain.md:1]: I merged incorrectly, fixed the issue now :)

File: packages/sit-onyx/src/components/OnyxForm/OnyxForm.stories.ts
Comment (JoCa96) [reply packages/sit-onyx/src/components/OnyxForm/OnyxForm.stories.ts:1]: Good catch :) But that is intentional. Forms will only provide the form injected props, but not inject them themselves.

**Pull request merged**